src/Uno.Toolkit.Skia.WinUI/Controls/Shadows/ShadowContainer.cs

```diff
@@ -252,6 +257,24 @@ protected override void OnApplyTemplate()
 
 		_shadowHost = skiaCanvas;
 		_canvas?.Children.Insert(0, _shadowHost!);
+
+		BindToShadowHost(_shadowHost);
+	}
+
+	private void BindToShadowHost(SKXamlCanvas skiaCanvas)
+	{
+		_shadowHostSubscriptions.Disposable?.Dispose();
```

Comment:
```suggestion
```
the assignment below automagically dispose the previous
unless you need make sure previous is cleared before installing a new one
there is no need to clear it before hand

Reply:
ok